pkgs/development/interpreters/jruby/default.nix

```diff
@@ -18,12 +18,28 @@ stdenv.mkDerivation rec {
      rm $out/bin/*.{bat,dll,exe,sh}
      mv $out/COPYING $out/LICENSE* $out/docs
 
-     for i in $out/bin/*; do
+     for i in $out/bin/jruby{,.bash}; do
        wrapProgram $i \
          --set JAVA_HOME ${jre}
      done
+
+     ln -s $out/bin/jruby $out/bin/ruby
+
+     # Bundler tries to create this directory
+     mkdir -pv $out/${gemPath}
+     mkdir -p $out/nix-support
+     cat > $out/nix-support/setup-hook <<EOF
+       addGemPath() {
+         addToSearchPath GEM_PATH \$1/${gemPath}
+       }
+
+       envHooks+=(addGemPath)
+     EOF
   '';
 
+  rubyEngine = "jruby";
+  gemPath = "lib/${rubyEngine}/gems/${version}";
```

Comment:
These need to be mover to the passthru as they are not used by the build but as attributes when passing the derivation around.